### PR TITLE
Unit Testing for Code Coverage

### PR DESCRIPTION
**Pull request by JT2M0L3Y**
## Added

* Vitest-based configuration for unit testing
* Unit tests for major non-discordjs classes, methods, and features

## Notes

* Discord.js mock testing of events, commands, and message passing is sorely lacking
* An open-source package, [gauntlet](https://github.com/ephemeralrogue/gauntlet) is being developed for this purpose

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,8 @@
     <h3><a href="#"></a>Ollama as your Discord AI Assistant</h3>
     <p><a href="#"></a><a href="https://creativecommons.org/licenses/by/4.0/"><img alt="License" src="https://img.shields.io/badge/License-CC_BY_4.0-darkgreen.svg" /></a>
     <a href="#"></a><a href="https://github.com/kevinthedang/discord-ollama/releases/latest"><img alt="Release" src="https://img.shields.io/github/v/release/kevinthedang/discord-ollama?logo=github" /></a>
-    <a href="#"></a><a href="https://github.com/kevinthedang/discord-ollama/actions/workflows/build-test.yml"><img alt="Build Status" src="https://github.com/kevinthedang/discord-ollama/actions/workflows/build-test.yml/badge.svg" /></a>
+    <a href="#"></a><a href="https://github.com/kevinthedang/discord-ollama/actions/workflows/build.yml"><img alt="Build Status" src="https://github.com/kevinthedang/discord-ollama/actions/workflows/build.yml/badge.svg" /></a>
```
Comment (kevinthedang) [inline README.md:7]: The image source is `Not Found`. `build-test.yml` still works as an image?

File: __tests__/queue.test.ts
Comment (kevinthedang) [inline __tests__/queue.test.ts:1]: File could use some comments. What if I don't understand why you made these tests?

File: __tests__/mentionClean.test.ts
Comment (kevinthedang) [inline __tests__/mentionClean.test.ts:1]: I guess the `it` describes the test. Looks good enough. I guess that goes the same for the queue tests but there are more there.

File: __tests__/getEnvVar.test.ts
Comment (kevinthedang) [inline __tests__/getEnvVar.test.ts:1]: Comments?

File: __tests__/events.test.ts
Comment (kevinthedang) [inline __tests__/events.test.ts:1]: Comments?

File: __tests__/commands.test.ts
Comment (kevinthedang) [inline __tests__/commands.test.ts:1]: I guess some comments would be nice...

**Review by kevinthedang — CHANGES_REQUESTED** [review]
The tests look fine. I haven't run them yet, but I hope they work.

URL link for Build is not working as intended as of now. Might fix itself later by running workflows.

**Comment by JT2M0L3Y** [comment]
> The tests look fine. I haven't run them yet, but I hope they work.
> 
> 
> 
> URL link for Build is not working as intended as of now. Might fix itself later by running workflows.

The url for build does not work in branches other than master. This is because the build workflow is only triggered upon a push to master (the result of merging this PR). I ran the existing tests many a time before pushing. Each existing test works with the current code base.

File: README.md (hunk 1 above)
Comment (JT2M0L3Y) [reply README.md:7]: I changed the file name the build.yml as I separated build and test into separate workflows.

File: __tests__/mentionClean.test.ts
Comment (JT2M0L3Y) [reply __tests__/mentionClean.test.ts:1]: Yes, the queue tests were weird to write. I don't think classes are normally tested in this format.

**Review by kevinthedang — APPROVED** [review]
A lot better and allows for easy understanding of what these do! Great work.

Will mention #14 for sake of logging because we don't want to close it just yet.

**Comment by kevinthedang** [comment]
Also I do not believe the version needs to be incremented because we are just introducing tests not a feature (unless you think we should then change it appropriately. @JT2M0L3Y

**Review by kevinthedang — CHANGES_REQUESTED** [review]
is there a reason why you're doing `__tests__/` instead of `tests/`?

Is there any way we can Mock the environment variables to reduce dependency on github? I think it's better to allow for isolation.

**Comment by JT2M0L3Y** [comment]
> is there a reason why you're doing `__tests__/` instead of `tests/`?

This is a convention from other NodeJS projects I have seen. However, there is a good chance this is an older styling convention and we could rather use just a folder named `/test` to model a UNIX-like file system. Would this be preferrable?

> Is there any way we can Mock the environment variables to reduce dependency on github? I think it's better to allow for isolation.

We could define a distinct, example set of environment variables that are compared during testing instead of using those passed into the `Keys` object by the workflow.

**Comment by kevinthedang** [comment]
Might even consider tests with their files but we shouldn't do that prob.

I guess it does not really matter per this [discussion](https://github.com/jestjs/jest/issues/637). But essentially, if we do not anticipate other "test" folders then we should even move the folder to `src/tests` but we can do 'tests/. I'm more down for the latter but let me know what you think.

**Review by kevinthedang — APPROVED** [review]
Looks great. For now the things that rely on Env variables can be there.

I suggest making an issue on Mocking those tests.

**Pull request merged**